builder/proxmox/common/step_start_vm.go

```diff
@@ -295,6 +296,26 @@ func generateProxmoxDisks(disks []diskConfig) proxmox.QemuDevices {
 	return devs
 }
 
+func generateProxmoxPCIDeviceMap(devices []pciDeviceConfig) proxmox.QemuDevices {
+	devs := make(proxmox.QemuDevices)
+	for idx := range devices {
+		devs[idx] = make(proxmox.QemuDevice)
+		setDeviceParamIfDefined(devs[idx], "host", devices[idx].Host)
+		setDeviceParamIfDefined(devs[idx], "device-id", devices[idx].DeviceID)
+		setDeviceParamIfDefined(devs[idx], "legacy-igd", strconv.FormatBool(devices[idx].LegacyIGD))
```

Comment:
As we discuss in #207, `setDeviceParamIfDefined` only behaves correctly when dealing with `string` values, and since these are boolean options I would suggest replacing those invocations by a `if`.

Reply:
Sure, but I'm not entirely convinced we should do it?

I don't think it's enough to just check if the proxmox default is false and then exclude the value if the bool is false. For example, a machine has set `legacy_igd=true`. You attempt to clone it, but decide against the flag so you set it to false in packer config in which case it would be just omitted and the result would not be what you expected. To me that is worse than perhaps somewhat redundantly displaying the default value in proxmox web interface. wdyt?

Reply:
Just to clarify, this is what I mean when I say "displaying default values in proxmox".

The ROM-Bar field defaults to `true`, so in proxmox when you add a PCI device it is not visible in the "hardware list view" (the line that starts with `hostpci0` behind the modal) but because we set the value to it's default when creating the VM - it gets appeneded to the config and thus also visible in the interface (even though it's at default value)

<img width="647" alt="image" src="https://github.com/hashicorp/packer-plugin-proxmox/assets/6570684/f19171c8-8e3e-44ca-99ab-96eaae63e7f1">

Reply:
In this case I would say this is a matter of intent, and how we want to convey it. `setDeviceParamIfDefined` implies that we set the boolean only if it's true (the function misbehaves with such attributes, but this is an alternate discussion to have).
Your description in my opinion warrants getting rid of both the function call, and the `if` then, and we should always set their value to the boolean's (or the inverse in the case of `HideROMBAR`), and not within `setDeviceParamIfDefined`, if only to make it clear this is intended to be part of the payload, regardless if it's been set or not.
Does that make sense?

Reply:
Yeah, I think so. Can you please take a look now?

Reply:
Looks good to me then!